src/Service/Resolvers/SqlQueryEngine.cs

```diff
@@ -213,6 +215,40 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
             // nextLink is the URL needed to get the next page of records using the same query options
             // with $after base64 encoded for opaqueness
             string path = UriHelper.GetEncodedUrl(_httpContextAccessor.HttpContext!.Request).Split('?')[0];
+
+            RuntimeConfig runtimeConfig = _runtimeConfigProvider.GetConfig();
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string? baseRoute = runtimeConfig.Runtime.Rest.BaseRoute;
+            if (!string.IsNullOrEmpty(baseRoute))
+            {
+                _sqlMetadataProvider.TryGetEntityPathFromName(context.EntityName, out string? pathNameForEntity);
```

Comment:
Can we move this up in the `if` condition so that we proceed only when we get a `pathNameForEntity`

Reply:
I am not sure what you mean here. The pathNameForEntity will either be the entity's name or entity's rest path. In any case, if the base route is non-empty (and obviously non-null), we would perform the same logic.

Reply:
```
if (!string.IsNullOrEmpty(baseRoute) && _sqlMetadataProvider.TryGetEntityPathFromName(context.EntityName, out string? pathNameForEntity))
{
 ...
} 
```
I mean the above code, this way you dont need to use null forgiving operator further below the code for `pathNameForEntity`

Reply:
Oh okay, got it. Don't need this code now though.